python/seldon_core/batch_processor.py

```diff
@@ -224,10 +256,133 @@ def _start_request_worker(
                 retries,
                 batch_id,
             )
+        # Mark task as done in the queue to add space for new tasks
         q_out.put(str_output)
         q_in.task_done()
 
 
+def _send_batch_predict_multi_request(
+    input_data: [],
+    data_type: str,
+    sc: SeldonClient,
+    retries: int,
+    batch_id: str,
+    payload_type: str,
+) -> str:
+    """
+    Send an request using the Seldon Client with batch context including the
+    unique ID of the batch and the Batch enumerated index as metadata. This
+    function also uses the unique batch ID as request ID so the request can be
+    traced back individually in the Seldon Request Logger context. Each request
+    will be attempted for the number of retries, and will return the string
+    serialised result. This method is similar to _send_batch_predict, but allows multiple
+    requests to be combined into a single prediction.
+    Parameters
+    ---
+    input_data
+        The input data containing the indexes, instance_ids and predictions
+    data_type
+        The data type to send which can be str, json and data
+    sc
+        The instance of SeldonClient to use to send the requests to the seldon model
+    retries
+        The number of times to retry the request
+    batch_id
+        The unique identifier for the batch which is passed to all requests
+    Returns
+    ---
+        A string serialised result of the response (or equivalent data with error info)
+    """
+
+    indexes = [x[0] for x in input_data]
+    instance_ids = [x[1] for x in input_data]
+    first_prediction = input_data[0][2]
+
+    predict_kwargs = {}
+    meta = {
+        "tags": {
+            "batch_id": batch_id,
+        }
+    }
+
+    predict_kwargs["meta"] = meta
+    predict_kwargs["headers"] = {"Seldon-Puid": instance_ids[0]}
+
+    try:
+        if data_type == "data":
+            # Initialise concatenated array for data
+            data = json.loads(first_prediction)
+            data_np = np.array(data)
+            concat = data_np
+            for i, raw_data in enumerate(input_data):
+                # Already added first item.
+                if i == 0:
+                    continue
+                data = json.loads(raw_data[2])
+                data_np = np.array(data)
+                concat = np.concatenate((concat, data_np))
+            predict_kwargs["data"] = concat
+        data = json.loads(first_prediction)
+        if data_type == "str":
```

Comment:
Why do we consider `str` and `json` here at all if the assumption is that this should work only with `data` on  `ndarray` and `tensor` payloads?

Reply:
Yeah good point, I can remove if we keep it as two functions, just a copy paste leftover